File: cameraview/src/main/java/com/base/cameraview/controls/WhiteBalance.java
package com.base.cameraview.controls;


import androidx.annotation.NonNull;

import com.base.cameraview.CameraOptions;

/**
 * White balance values control the white balance settings.
 */
public enum WhiteBalance implements Control {

    /**
     * Automatic white balance selection (AWB).
     * This is not guaranteed to be supported.
     *
     * @see CameraOptions#getSupportedWhiteBalance()
     */
    AUTO(0),

    /**
     * White balance appropriate for incandescent light.
     * This is not guaranteed to be supported.
     *
     * @see CameraOptions#getSupportedWhiteBalance()
     */
    INCANDESCENT(1),

    /**
     * White balance appropriate for fluorescent light.
     * This is not guaranteed to be supported.
     *
     * @see CameraOptions#getSupportedWhiteBalance()
     */
    FLUORESCENT(2),

    /**
     * White balance appropriate for daylight captures.
     * This is not guaranteed to be supported.
     *
     * @see CameraOptions#getSupportedWhiteBalance()
     */
    DAYLIGHT(3),

    /**
     * White balance appropriate for pictures in cloudy conditions.
     * This is not guaranteed to be supported.
     *
     * @see CameraOptions#getSupportedWhiteBalance()
     */
    CLOUDY(4);

    static final WhiteBalance DEFAULT = AUTO;

    private int value;

    WhiteBalance(int value) {
        this.value = value;
    }

    @NonNull
    static WhiteBalance fromValue(int value) {
        WhiteBalance[] list = WhiteBalance.values();
        for (WhiteBalance action : list) {
            if (action.value() == value) {
                return action;
            }
        }
        return DEFAULT;
    }

    int value() {
        return value;
    }
}

File: cameraview/src/main/java/com/base/cameraview/controls/WhiteBalance.kt
package com.base.cameraview.controls


/**
 * White balance values control the white balance settings.
 */
enum class WhiteBalance(val value: Int) : Control {
    /**
     * Automatic white balance selection (AWB).
     * This is not guaranteed to be supported.
     *
     * @see com.base.cameraview.CameraOptions.getSupportedWhiteBalance
     */
    AUTO(0),

    /**
     * White balance appropriate for incandescent light.
     * This is not guaranteed to be supported.
     *
     * @see com.base.cameraview.CameraOptions.getSupportedWhiteBalance
     */
    INCANDESCENT(1),

    /**
     * White balance appropriate for fluorescent light.
     * This is not guaranteed to be supported.
     *
     * @see com.base.cameraview.CameraOptions.getSupportedWhiteBalance
     */
    FLUORESCENT(2),

    /**
     * White balance appropriate for daylight captures.
     * This is not guaranteed to be supported.
     *
     * @see com.base.cameraview.CameraOptions.getSupportedWhiteBalance
     */
    DAYLIGHT(3),

    /**
     * White balance appropriate for pictures in cloudy conditions.
     * This is not guaranteed to be supported.
     *
     * @see com.base.cameraview.CameraOptions.getSupportedWhiteBalance
     */
    CLOUDY(4);

    companion object {
        @JvmField
        val DEFAULT: WhiteBalance = AUTO

        @JvmStatic
        fun fromValue(value: Int): WhiteBalance {
            val list = WhiteBalance.entries.toTypedArray()
            for (action in list) {
                if (action.value == value) {
                    return action
                }
            }
            return DEFAULT
        }
    }
}